From cf38eb874997e6fad3aa832d05a2f08783e635d6 Mon Sep 17 00:00:00 2001
From: Tamal <tasaha@habana.ai>
Date: Wed, 21 Aug 2024 07:38:24 +0300
Subject: [PATCH] [SW-161727] g3: remove global variable for bist

This patch removes global variables used in
ppboot for bist.

Change-Id: I1a4754b341f8688b7344d6048d27fb0e3823c558
Signed-off-by: Tamal <tasaha@habana.ai>
---

diff --git a/arch/gaudi3/arc/asic.c b/arch/gaudi3/arc/asic.c
index 6140eeb..50cc976 100644
--- a/arch/gaudi3/arc/asic.c
+++ b/arch/gaudi3/arc/asic.c
@@ -33,26 +33,24 @@
 #include <psoc_reset_conf_regs.h>
 #include <psoc_reset_conf_masks.h>
 
-static struct hl_ras_sram_bist_data bist_data;
-static u32 *hl_gaudi3_sram_mbist_data_start_address;
-
-static void gaudi3_sram_self_repair(u64 sram_loadaddr)
+static void gaudi3_sram_self_repair(u64 sram_loadaddr, struct hl_ras_sram_bist_data *bist_data)
 {
+	u32 *hl_gaudi3_sram_mbist_data_start_address;
 	int status = 0;
 
 	hl_gaudi3_sram_mbist_data_start_address = (u32 *) ((u32)((sram_loadaddr &
 				ARC_ADDR_MASK) | FW_MEM_REGION_BASE));
 
 	status = hl_ppboot_sram_self_repair_mismatch_logic(mmD0_PSOC_DFT_APB_BASE,
-		hl_gaudi3_sram_mbist_data_start_address, &bist_data,
+		hl_gaudi3_sram_mbist_data_start_address, bist_data,
 		SRAM_MBIST_FAIL_CNTRL_MAX_COUNT);
 
-	bist_data.magic_number = SRAM_BIST_MAGIC;
+	bist_data->magic_number = SRAM_BIST_MAGIC;
 	if (status) {
-		bist_data.bist_result = SRAM_MBIST_DIE0_CACHE_FAIL;
+		bist_data->bist_result = SRAM_MBIST_DIE0_CACHE_FAIL;
 		preboot_puts("Die 0 cache memories MBIST fail\n");
 	} else {
-		bist_data.bist_result = SRAM_MBIST_ALL_PASS;
+		bist_data->bist_result = SRAM_MBIST_ALL_PASS;
 		preboot_puts("Die 0 cache memories MBIST Pass\n");
 	}
 }
@@ -202,12 +200,12 @@
 	udelay(1000);
 }
 
-static inline void gaudi3_write_sram_bist_result(void)
+static inline void gaudi3_write_sram_bist_result(void *bist_data)
 {
 	/* Store Ppboot MBIST result in boot info */
 
 	if (ppboot_boot_info_set(HL_BOOT_INFO_SRAM_BIST_RESULT,
-				(void *)&bist_data,
+				bist_data,
 				sizeof(struct hl_ras_sram_bist_data)))
 		preboot_puts("BOOT_INFO: Failed to set sram mbist result\n");
 	else
@@ -217,6 +215,7 @@
 void hl_sram_self_repair(void)
 {
 	struct gaudi3_cold_rst_data cold_rst_data;
+	struct hl_ras_sram_bist_data bist_data;
 	u64 sram_loadaddr;
 	bool ret;
 
@@ -260,10 +259,10 @@
 		return;
 	}
 
-	gaudi3_sram_self_repair(sram_loadaddr);
+	gaudi3_sram_self_repair(sram_loadaddr, &bist_data);
 	gaudi3_die0_cs_reset();
 	gaudi3_sram_ecc_init();
-	gaudi3_write_sram_bist_result();
+	gaudi3_write_sram_bist_result((void *)&bist_data);
 	preboot_puts("bist done for 48MB SRAM/CACHE\n");
 }
 
